src/Symfony/Bundle/FrameworkBundle/Command/ServerCommand.php

```diff
@@ -41,4 +41,25 @@ protected function getLockFile($address)
     {
         return sys_get_temp_dir().'/'.strtr($address, '.:', '--').'.pid';
     }
+
+    protected function isOtherServerProcessRunning($address)
+    {
+        $lockFile = $this->getLockFile($address);
+
+        if (file_exists($lockFile)) {
+            return true;
+        }
+
+        list($hostname, $port) = explode(':', $address);
```

Comment:
To not do this several times, we could split the given address in the command itself and pass the host and the ip address separately to this method.